lib/grape/validations/validators/coerce.rb

```diff
@@ -54,7 +54,14 @@ def coerce_value(type, val)
         return val || Set.new if type == Set
         return val || {}      if type == Hash
 
-        converter = Virtus::Attribute.build(type)
+        # To support custom types that Virtus can't easily coerce, pass in an
+        # explicit coercer. Custom types must implement a `parse` class method.
+        converter_options = {}
+        if ![DateTime, Date, Time].include?(type) && type.respond_to?(:parse)
```

Comment:
So I haven't been able to find a "good" way to identify if Virtus knows about a type or not... this works, but I think it relies too much on internal knowledge to be usable:

```ruby
def virtus_type?(type)
  Coercible::Coercer.const_get(type.to_s, false) 
end
```

However, it occurs to me that Grape doesn't have any single place where the parameter types you can use are listed. Nowhere in the README is there a list of what types one can use; it's just shown by example that String, Integer, Float, Date, and DateTime are the useful ones. What if we made the list of "default types" explicit, and simply checked whether or not to coerce by looking at our list of default types?

The added benefit of doing that is that we could detect invalid parameter types at runtime. I initially tried to add an error along the lines of `you're using a custom type but it doesn't respond to :parse` to this PR, but was stymied by the fact most of the default types would trigger that error. If we could immediately identify that a custom type is being used, we could add more helpful error messages and make things easier on developers.

I'm thinking something along the lines of `Grape::ParameterTypes`, ala `Grape::ContentTypes`. What do you think?